### PR TITLE
[flink] fix the problem that DynamicFilteringEvent does not exist in version 1.16- of flink.

### LINKED ISSUE #2304: [Bug] BatchFileStoreITCase always fails.
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

paimon: master

### Compute Engine

flink 

### Minimal reproduce step

Run `BatchFileStoreITCase#testOverwriteEmpty` in `paimon-flink-1.14` and `paimon-flink-1.15`.

### What doesn't meet your expectations?

`DynamicFilteringEvent` comes from `flink 1.16+` version, which causes `BatchFileStoreITCase#testOverwriteEmpty `in `flink-1.14` and `flink-1.15` to always fail.

https://github.com/apache/incubator-paimon/actions/runs/6846772528/job/18613902157?pr=2221

### Anything else?

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by liming30**
### Purpose

Linked issue: close #2304 

[flink] fix the problem that `DynamicFilteringEvent` does not exist in version `1.16-` of flink.

### Tests

Existing tests can cover this change.

### API and Format

No
### Documentation

No

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Can you add `BatchFileStoreITCase` to 1.14 and 1.15?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java [hunk 1]
```diff
@@ -65,59 +49,10 @@ public StaticFileStoreSplitEnumerator(
...
-    }
-
-    @Nullable
-    public Snapshot snapshot() {
-        return snapshot;
-    }
-
     @Override
     public void handleSourceEvent(int subtaskId, SourceEvent sourceEvent) {
         if (sourceEvent instanceof DynamicFilteringEvent) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java:58]: Can we modify this to `if (sourceEvent != null && sourceEvent.getClass().getName().endsWith("DynamicFilteringEvent"))`?

And extract DynamicFiltering logical from here to a separate class?

**Comment by JingsongLi** [comment]
Thanks @liming30 , left two comments.

**Comment by liming30** [comment]
> Can you add `BatchFileStoreITCase` to 1.14 and 1.15?

`BatchFileStoreITCase` already exists in 1.14 and 1.15.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java (hunk 1 above)
Comment (liming30) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java]: Sounds great, I have addressed it according to the comment.

**Comment by JingsongLi** [comment]
In #2304 , 
```
Minimal reproduce step
Run BatchFileStoreITCase#testOverwriteEmpty in paimon-flink-1.14 and paimon-flink-1.15.
```
What do you mean?

**Comment by liming30** [comment]
> In #2304 ,
> 
> ```
> Minimal reproduce step
> Run BatchFileStoreITCase#testOverwriteEmpty in paimon-flink-1.14 and paimon-flink-1.15.
> ```
> 
> What do you mean?

My mistake. In #2221, I introduced `ReaderConsumeProgressEvent`.  `StaticFileStoreSplitEnumerator` will go into the `handleSourceEvent` method after split is consumed. At this time,  `java.lang.ClassNotFoundException: org.apache.flink.table.connector.source.DynamicFilteringEvent` will appear in flink-1.14 and flink-1.15.

Do you think the changes here should be moved to #2221?

**Review by JingsongLi — APPROVED** [review]
+1

**Comment by JingsongLi** [comment]
@liming30 It is OK to merge this.

**Pull request merged**